docs/scaling/README.md

```diff
@@ -0,0 +1,47 @@
+# Autoscaling
+
+There are three main components to the autoscaling system: autoscaler, activator and queue-proxy.
+
+## Queue-proxy
+Whenever a Serverless Service is deployed by the user, the queue-proxy sidecar container is injected with it. This queue proxy does the following:
+ - Any request coming to the application user-container will go to the queue-proxy and then be routed to the user-container
+ - The readiness and liveness probes defined by the user will be replaced with queue-proxy probes externally to the pod while the user-containers probe endpoints will only be accessed by the proxy.
+ - Makes sure that no more than the ‘defined container concurrency’ requests reach the application's instance at once by queueing other requests, for example if a revision defines a concurrency limit of 5, the queue-proxy makes sure that no more than 5 requests reach the application's instance at once. If there are more requests being sent to it than that, it will queue them locally.
```

Comment:
```suggestion
 - Makes sure that no more than the ‘defined container concurrency’ requests reach the application's instance at once by queueing other requests. For example if a revision defines a concurrency limit of 5, the queue-proxy makes sure that no more than 5 requests reach the application's instance at once. If there are more requests being sent to it than that, it will queue them locally.
```